sdk/iOS/src/MSSyncContext.h

```diff
@@ -101,7 +102,7 @@
 @property (nonatomic, readonly) NSUInteger pendingOperationsCount;
 
 /// Executes all current pending operations on the queue
-- (void) pushWithCompletion:(MSSyncBlock)completion;
+- (MSQueuePushOperation *) pushWithCompletion:(MSSyncBlock)completion;
```

Comment:
I don't generally need to cancel, but you could.

My main use case would be adding other operations as dependencies when calling multiple `pullWithQuery`. i.e. I loop through tables and pull on each. If the actual operation was returned, I then have another NSOperation instance which I can add all of these pulls as dependencies of said operation.

Currently they are project scoped. I get around this, as I am importing the source via cocoa pods and so can access all the headers. If you agree that returning the operations will be acceptable, I can make those headers public.